### ISSUE TITLE
At run time signalling channel break and try to reconnect. While client connection state still connected it return and NO AV frame on viewer

### ISSUE THREAD

**Issue by shiv50084** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1074]
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/samples/Common.c#L1196

We facing this issue when master signaling channel disconnect and try to connect again and while viewer state still connected.  at that time we received offer again in  API "signalingMessageReceived". It return from API given API from LINE "https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/samples/Common.c#L1196"

and viewer show black frame while we sending AV frames from master.

I want to is there any need to free previous signaling channel  data to connect with new one. May this is the issue. How to resolve that and which API i need to free previous state data and use new one

**SDK version number**
latest master commit: f3a34c21ab6018074607fc3cb3ac3701314bee67

**Comment by MushMal** [comment]
I am having a hard time understanding your scenario. Can you rephrase it? Who is sending a new offer? If your channel connection is broken then likely there is a network issue. Try to debug this further

**Comment by disa6302** [comment]
@shiv50084 ,

Any updates?

**Comment by disa6302** [comment]
Closing due to inactivity.
